### PR TITLE
Add `default.tensor` to `setup.py`

### PR DESCRIPTION
**Pull request by mudit2812**
As name says.

* `default.tensor` is the updated name of the tensor network device with the `quimb` backend. The new change installs it along with `lightning.qubit` as there are no non-python dependencies to get the device to work.
* Updated the device's `name` attribute to return `default.tensor` instead of `lightning.tensor`.
* Updated the errors raised by the device to use `default.tensor` instead of `LightningTensor` or `lightning.tensor`.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@
 * Add support for `qml.expval` and `qml.var` in the `lightning.tensor` device for the `quimb` interface and the MPS method.
   [(#686)](https://github.com/PennyLaneAI/pennylane-lightning/pull/686)  
 
+* Changed the name of `lightning.tensor` to `default.tensor` with the `quimb` backend.
+  [(#)]()
```
Comment (mudit2812) [inline .github/CHANGELOG.md:19]:
```suggestion
  [(#719)](https://github.com/PennyLaneAI/pennylane-lightning/pull/719)
```

File: setup.py [hunk 2]
```diff
@@ -187,6 +187,13 @@ def build_extension(self, ext: CMakeExtension):
 suffix = suffix[0].upper() + suffix[1:]
 
 pennylane_plugins = [device_name + " = pennylane_lightning." + backend + ":Lightning" + suffix]
+if suffix == "Qubit":
+    print(
+        "&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&"
+    )
```
Comment (mudit2812) [inline setup.py:193]:
```suggestion
```

**Review by mlxd — COMMENTED** [review]
Thanks Mudit --- can we confirm when building this as a wheel the new folders/files are also included?

File: setup.py
Comment (mlxd) [inline setup.py:1]: To ensure the new files are also included with the wheel builds, you can modify L180 to add `pennylane_lightning.lightning_tensor` and then also at L255 and below to include the new folders/files for that added packages with 
```python
                "pennylane_lightning.lightning_tensor": [
                    os.path.join("backends", "*"),
                    os.path.join("backends", "**", "*"),
                ],
```

**Comment by mudit2812** [comment]
[sc-61741]

File: setup.py [hunk 3]
```diff
@@ -187,6 +187,10 @@ def build_extension(self, ext: CMakeExtension):
 suffix = suffix[0].upper() + suffix[1:]
 
 pennylane_plugins = [device_name + " = pennylane_lightning." + backend + ":Lightning" + suffix]
+if suffix == "Qubit":
```
Comment (PietropaoloFrisoni) [inline setup.py:190]: I can't help but think it's funny that in the end we only needed these 4 lines in `setup.py`. Except for the changes that @mlxd requested, for me this is good to go

**Review by PietropaoloFrisoni — APPROVED** [review]

**Comment by PietropaoloFrisoni** [comment]
[This](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/9007812781/job/24748695976?pr=719) is obviously a statistical fluctuation. Codecov keeps complaining, but at this point I am not surprised

**Review by vincentmr — COMMENTED** [review]
Do we want to replace the following too?
```
pennylane_lightning/lightning_tensor/backends/quimb/_mps.py:    """A function that determines if an operation is supported by ``lightning.tensor`` for this interface."""
pennylane_lightning/lightning_tensor/backends/quimb/_mps.py:    """A function that determines if an observable is supported by ``lightning.tensor`` for this interface."""
pennylane_lightning/lightning_tensor/lightning_tensor.py:    """A function that determines whether or not a backend is supported by ``lightning.tensor``."""
pennylane_lightning/lightning_tensor/lightning_tensor.py:    """A function that determines whether or not a method is supported by ``lightning.tensor``."""
```

**Comment by mudit2812** [comment]
> Do we want to replace the following too?
> 
> ```
> pennylane_lightning/lightning_tensor/backends/quimb/_mps.py:    """A function that determines if an operation is supported by ``lightning.tensor`` for this interface."""
> pennylane_lightning/lightning_tensor/backends/quimb/_mps.py:    """A function that determines if an observable is supported by ``lightning.tensor`` for this interface."""
> pennylane_lightning/lightning_tensor/lightning_tensor.py:    """A function that determines whether or not a backend is supported by ``lightning.tensor``."""
> pennylane_lightning/lightning_tensor/lightning_tensor.py:    """A function that determines whether or not a method is supported by ``lightning.tensor``."""
> ```

Where is this from?

**Comment by vincentmr** [comment]
> Where is this from?

I was looking for the `lightning.tensor` string in files with `grep -r "lightning\.tensor" pennylane_lightning/`.

**Review by mlxd — APPROVED** [review]
No more concerns from my side. Thanks @mudit2812 
Feel free to merge once all CI runs are green.

**Review by vincentmr — APPROVED** [review]
Looks good to me, thanks @mudit2812 .

**Pull request merged**